packages/@sfcc-bff/basketapi/src/api/schema/basketResolvers.ts

```diff
@@ -121,19 +121,25 @@ const getBasket = async (config: Config, context: AppContext) => {
             productIds,
         );
         // update the image to the product items in basket
-        basket.productItems.map(product => {
-            productDetails.find(item => {
-                if (item.pid === product.productId) {
-                    product.image = item.imageURL ? item.imageURL : '';
-                }
-            });
+        basket.productItems.forEach(product => {
+            const item = productDetails.find(
+                item => item.pid === product.productId,
```

Comment:
@gurpreetsainisalesforce you can't do `break` inside a `forEach`. I changed the code to be more semantic and properly use a `.find`.